### ISSUE TITLE
strutils:formatSize() is broken in JS mode

### ISSUE THREAD

**Issue by genotrance** [nim-lang/Nim#8231]
```nim
import strutils

echo formatSize(2462056448, '.', bpIEC, false)
```

Echos 2.293GiB for C/C++ mode but 37568YiB for JS mode. Fails test case in [strutils.nim](https://github.com/nim-lang/Nim/blob/8667899d531225e060db3effe72627b9609017ce/lib/pure/strutils.nim#L2404) but isn't getting caught since testament doesn't actually run the test, just compiles due to separate bug.

**Comment by ghost** [comment]
I've investigated a bit, and it seems to be like that because bitwise operators in JS are [32-bit](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators), so `1 << 40` gives `256` and not `1099511627776`
A possible fix might be to make custom JS functions for bitwise operations so they'll work on both 32-bit and 64-bit

**Comment by mratsim** [comment]
Fixed by #15070
